### PR TITLE
Converted the Entire codebase into typescript

### PR DESCRIPTION
**Pull request by KayzeTheGeek**
- I made some fields that were not optional for the connect, consumer, and producer functions optional
- Changed "module.export" to "export default"
- Configured the tsconfig.json file
-  Configured the package.json file
- Updated some parts of the readme

### CODE REVIEW

**Comment by KayzeTheGeek** [comment]
So I have updated it you can review now

**Review by idanasulinmemphis — DISMISSED** [review]

File: examples/consumer.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import memphis from "memphis-dev"
+
+
```
Comment (idanasulinmemphis) [inline examples/consumer.ts:3]: extra line break

File: examples/producer.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            stationName: "<station-name>",
+            producerName: "<producer-name>"
+        });
+
+        for (let index = 0; index < 100; index++) {
+            await producer.produce({
+                message: Buffer.from(`Message #${index}: Hello world`)
+            });
+        }
+
```
Comment (idanasulinmemphis) [inline examples/producer.ts:21]: extra line break

File: src/index.ts [hunk 3]
```diff
@@ -13,8 +13,7 @@
 
 import net from 'net';
 import events from 'events';
-import * as broker from 'nats';
-import { headers } from "nats";
+import broker, { headers } from 'nats';
```
Comment (idanasulinmemphis) [inline src/index.ts:16]: please remove and return to the original import

**Pull request closed without merging**